internal/oci/oci_linux.go

```diff
@@ -83,45 +85,47 @@ func newPipe() (parent, child *os.File, err error) {
 	return os.NewFile(uintptr(fds[1]), "parent"), os.NewFile(uintptr(fds[0]), "child"), nil
 }
 
-func loadFactory(root string) (libcontainer.Factory, error) {
-	abs, err := filepath.Abs(root)
+func (r *runtimeOCI) containerStats(ctr *Container) (*ContainerStats, error) {
+	eventStats, err := r.eventStats(ctr)
 	if err != nil {
 		return nil, err
 	}
-	cgroupManager := libcontainer.Cgroupfs
-	return libcontainer.New(abs, cgroupManager, libcontainer.CriuPath(""))
+	stats := &ContainerStats{}
+	stats.Container = ctr.ID()
+	stats.CPUNano = eventStats.CPU.Usage.Total
+	stats.SystemNano = time.Now().UnixNano()
+	stats.CPU = calculateCPUPercent(eventStats)
+	stats.MemUsage = eventStats.Memory.Usage.Usage
+	stats.MemLimit = getMemLimit(eventStats.Memory.Usage.Limit)
+	stats.MemPerc = float64(stats.MemUsage) / float64(stats.MemLimit)
+	stats.PIDs = eventStats.Pids.Current
+	stats.BlockInput, stats.BlockOutput = calculateBlockIO(eventStats)
+	stats.NetInput, stats.NetOutput = getContainerNetIO(eventStats)
+
+	return stats, nil
 }
 
-// libcontainerStats gets the stats for the container with the given id from runc/libcontainer
-func (r *runtimeOCI) libcontainerStats(ctr *Container) (*libcontainer.Stats, error) {
-	factory, err := loadFactory(r.root)
+// eventStats gets the stats for the container with the given id from an OCI runtime
+func (r *runtimeOCI) eventStats(ctr *Container) (*types.Stats, error) {
+	res, err := utils.ExecCmd(r.path, rootFlag, r.root, "events", "--stats", ctr.ID())
```

Comment:
I don't think we can assume the OCI runtime has the `events` command.  It will work only with runc.

Reply:
Hm, does creating a libcontainer factory work with other runtimes, like crun?

Reply:
I thought so :( but looking at the code, it seems it is already broken.

We should read this information directly from the cgroups (possibly using github.com/containers/libpod/pkg/cgroups) and not depend from a single OCI runtime.

Reply:
Yeah, that would be an option. If I change this now we would definitely touch it again when we migrate to libpod.

Please take a look at: https://github.com/cri-o/cri-o/pull/3030